src/components/Avatar/index.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { ReactNode } from 'react';
+import PropTypes from 'prop-types';
+import StyledAvatar from './styled/avatar';
+import StyledImage from './styled/image';
+import StyledText from './styled/text';
+import StyledIcon from './styled/icon';
+import RenderIf from '../RenderIf';
+import { BaseProps } from '../types';
+
+type AvatarSizes = 'large' | 'medium' | 'small' | 'x-small';
+
+interface Props extends BaseProps {
+    icon?: ReactNode;
+    initial?: string;
+    src?: string;
+    size?: AvatarSizes;
+    style?: object;
+}
+
+const Avatar: React.FC<Props> = props => {
+    const { icon, initial, src, size, style } = props;
+    const isImage = !!src;
+    const isIcon = !!(icon !== null && src === '');
+    const isInitial = !!(typeof initial !== undefined && src === '');
```

Comment:
The components should not break when the user pass an invalid type to the props, then we need to check that initials and src are string types:
```const isInitial = !!(initial && typeof initials ==='string' && !src || typeof src !== 'string');```
Then we can put this in a helper function (showInitials) and test it in isolation:
```const isInitial = showInitialts({ initials, src });```

Reply:
Or previously checking the src type...
```
const isImage = !!(src && typeof src === 'string');
const isInitials = !!(initials && typeof initials === 'string' && !isImage);
```